src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -55,16 +91,57 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   // Send a user to datastore. Update the current information about the user if one exists.
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userGitHubID = request.getParameter("githubID");
-    if (userGitHubID.trim().isEmpty()) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid ID");
+    String userGitHubAuthToken;
+    ArrayList<String> userTags;
+    // Get information from request body
+    try {
+      userGitHubAuthToken = request.getParameter("gitHubAuthToken");
+      userTags = new ArrayList<>(Arrays.asList(request.getParameterValues("interestTags")));
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage() + "/n/n" + e.getStackTrace() + "/n/n Incomplete POST body parameters",
+          HttpServletResponse.SC_BAD_REQUEST,
+          "User information was incomplete");
+      return;
+    }
```

Comment:
Why is this in a try-catch block? [No error is ever thrown](https://tomcat.apache.org/tomcat-5.5-doc/servletapi/javax/servlet/ServletRequest.html#getParameter(java.lang.String)) from `request.getParameter("gitHubAuthToken");` and we could avoid an error being thrown in `userTags = new ArrayList<>(Arrays.asList(request.getParameterValues("interestTags")));` by checking first if the parameter is null and sending an error response then.